MHArmory.Search/OpenCL/OpenCLSearch.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MHArmory.Search.OpenCL
+{
+    public class OpenCLSearch
+    {
+        private SearchDataSerializer Serializer { get; }
+        private SearchResultDeserializer Deserializer { get; }
+        private Host Host { get; }
+
+        private static OpenCLSearch _instance;
+        public static OpenCLSearch Instance
+        {
+            get
+            {
+                return _instance ?? (_instance = new OpenCLSearch());
+            }
+        }
+
+        private OpenCLSearch()
+        {
+            Serializer = new SearchDataSerializer();
+            Deserializer = new SearchResultDeserializer();
+            Host = new Host();
+        }
+
+        // For debugging purposes
+        private void WriteByteArr(StreamWriter writer, string title, byte[] data)
```

Comment:
Please avoid short names, prefer `WriteByteArray` to `WriteByteArr`, even if this is just a debug method.